app/Http/Controllers/Forum/PostsController.php

```diff
@@ -32,35 +32,39 @@ class PostsController extends Controller
     public function __construct()
     {
         $this->middleware('auth', ['only' => [
-            'destroy',
+            'changeVisibility',
             'raw',
         ]]);
 
         return parent::__construct();
     }
 
-    public function destroy($id)
+    public function changeVisibility($id)
     {
-        $post = Post::findOrFail($id);
+        $showDeleted = priv_check('ForumTopicModerate')->can();
 
-        priv_check('ForumPostDelete', $post)->ensureCan();
+        $post = Post::query()->showDeleted($showDeleted)->findOrFail($id);
+        $topic = $post->topic()->showDeleted($showDeleted)->first();
 
-        $deletedPostPosition = $post->topic->postPosition($post->post_id);
+        $action = Request::input('action');
 
-        $post->topic->removePost($post, Auth::user());
+        priv_check('ForumPostDelete', $post)->ensureCan();
```

Comment:
this is a bit confusing because the position here means restore action also checks for `Delete`.